tests/test.rs

```diff
@@ -0,0 +1,105 @@
+use insta;
+use os_pipe::pipe;
+use std::io::Read;
+use std::sync::Mutex;
+use structopt::lazy_static::lazy_static;
+
+lazy_static! {
+    /// rust will try to run the tests in parallel by default, and `insta` doesn't like the
+    /// usual way of disabling this via `--test-threads=1`, so we're using this
+    /// mutex to make sure we're not re-flashing until the last run is finished
+    static ref ONE_RUN_AT_A_TIME: Mutex<i32> = Mutex::new(0i32);
```

Comment:
from what I gather, that's some [`lazy_static`](https://docs.rs/lazy_static/1.4.0/lazy_static/) trickery: We need the mutex to be "lazy static" because the Mutex is only created at runtime (whereas normally, only items that can be evaluated at compile time can be "regular" `static`). So our mutex is a [`ref`](https://doc.rust-lang.org/rust-by-example/scope/borrow/ref.html), e.g. a reference to a `Mutex` that is then  initialized when it is dereferenced for the first time (`lazy_static` implements this for us). Not 100% sure why this is required to be a `ref` rather than `static ONE_RUN_AT_A_TIME: &Mutex =  Mutex::new(0i32);` e.g. if that's either an implementation choice by `lazy_static` (and why it was made) or if that was the only way to go

Reply:
Thank you for the elaboration! 😄